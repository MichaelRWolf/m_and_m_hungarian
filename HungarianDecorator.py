"""
Module for decorating text with Hungarian notation.
"""
import nltk


class HungarianDecorator:
    def decorate_from_file(self, filename):
        pass

    def decorate_text(self, text):
        pass

    def decorate_sentence(self, sentence):
        tokens = nltk.word_tokenize(sentence)
        tagged = nltk.pos_tag(tokens)
        decorated_sentence = self.decorate_token(tagged)
        return decorated_sentence

    def decorate_token(self, token):
        decorated_token = []
        for word, tag in token:
            decorated_word = self.decorate_word(word)
            decorated_token.append(decorated_word)
        return decorated_token

    @staticmethod
    def decorate_word(word):
        return f"__{word}__"
        pass

    def decorate_punctuation(self, token):
        pass
