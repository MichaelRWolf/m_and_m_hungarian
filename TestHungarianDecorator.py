"""
Module to test the hungarian_decorator module.
"""

import unittest

import approvaltests

from HungarianDecorator import HungarianDecorator


class TestHungarianDecorator(unittest.TestCase):
    def setUp(self):
        self.decorator = HungarianDecorator()


def test_decorate_from_file():
    pass

    def test_decorate_text(self):
        pass

    def test_decorate_sentence(self):
        sentence = "This is a sentence."
        expected = "This is a sentence."
        actual = self.decorator.decorate_sentence(sentence)
 #       approvaltests.verify(actual, "received.txt")
        approvaltests.verify(actual)

    def test_decorate_token(self, word=""):
        pass

    def test_decorate_word(self):
        pass

    def test_decorate_punctuation(self):
        approvaltests.verify("Stuff")
