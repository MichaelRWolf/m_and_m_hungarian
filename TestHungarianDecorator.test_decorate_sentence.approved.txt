['__This__', '__is__', '__a__', '__sentence__', '__.__']
